tests/phpunit/Fixtures/SimpleMutationsCollectorVisitor.php

```diff
@@ -41,9 +42,9 @@ public function leaveNode(Node $node)
             return;
         }
 
-        $mutatedResult = $this->mutator->mutate($node);
-
-        $mutatedNodes = $mutatedResult instanceof Generator ? $mutatedResult : [$mutatedResult];
+        // It is important to not rely on the keys here. It might otherwise result in some elements
+        // being overridden, see https://3v4l.org/JLN73
+        $mutatedNodes = iterator_to_array($this->mutator->mutate($node), false);
```

Comment:
In this case `foreach ($generator as $val)` will keep all values regardless of keys, so you don't have to convert it to an array.

Reply:
Yes but we use the index: so we either do that or have a `$i = 0` that we increment. In both scenarios I think it's non-obvious at first glance and warrants a comment. I would love a test as well, but that's out of scope for now

Reply:
We don't here.

Reply:
Something like below should work just fine:
```
 foreach ($this->mutator->mutate($node) as $mutatedNode) {
```